packages/sdk/src/swap/sdk.ts

```diff
@@ -72,9 +74,22 @@ export class SwapSDK {
 
   executeSwap(params: ExecuteSwapParams): Promise<ContractReceipt> {
     assert(this.signer, 'No signer provided');
-    return executeSwap(params, {
-      network: this.network,
+    return executeSwap(params, { network: this.network, signer: this.signer });
+  }
+
+  async approveVault(
```

Comment:
should we expose a method like `needsVaultApproval`?

Reply:
+1 but could be its own ticket.